### DISCUSSION TITLE
EZO chip - Command not understood - bad detection after a soft reset

### DISCUSSION THREAD

**Discussion by SimonFili** [arendst/Tasmota#19785] [Support request] [answered]
Hi,

I'm getting this red flashing LED on an ORP EZO chip with a soft reset (from the web interface)
![image](https://github.com/arendst/Tasmota/assets/7669537/66e45965-0923-44c9-b4b9-78696160954a)
In that case, this EZO ORP does not appear on the main page.

On a hard reset (power on), all is fine.

When I get this red LED, I can still see the address with i2cscan, I can even interrogate with Sensor78 -2 i
```
07:34:09.207 CMD: i2cscan
07:34:09.226 RSL: RESULT = {"I2CScan":"Device(s) found on bus1 at 0x51 0x61 0x62 0x63 0x64 0x65 0x66"}

07:35:04.633 CMD: sensor78 -2 i
07:35:05.157 RSL: RESULT = {"Sensor":"?I,ORP,2.11"}
```
I have a total of 6 EZO chip, 4 ORP, 1 pH and 1 EC.

All is good and working on a hard reset

![image](https://github.com/arendst/Tasmota/assets/7669537/6582b482-f32d-4870-97af-eb0070771eae)
This is connected on a Lilygo ESP32 S3 relay board.
![image](https://github.com/arendst/Tasmota/assets/7669537/fc624db7-85d8-4af5-88ed-805c456b784f)

Here's the output from a soft reset:
We can see here that Tasmota detect an SCD40 (CO2 sensor) on 0x62 which I don't have.
So the issue seems to be an order of detection?  
Maybe I can remove the support from this chip at compile time?
Or change the i2c address of this ORP to something else?

```
00:00:00.002 HDW: ESP32-S3 
00:00:00.023 UFS: FlashFS mounted with 12592 kB free
00:00:00.037 CFG: Loaded from File, Count 560
00:00:00.045 QPC: Count 1
00:00:00.054 I2C: Bus1 using GPIO17(SCL) and GPIO16(SDA)
00:00:00.061 TFS: File 'mcp23x.dat' not found
00:00:00.116 BRY: Berry initialized, RAM used=4438 bytes
00:00:00.129 TFS: File '.drvset003' not found
00:00:00.140 Project tasmota - Tasmota Version 13.1.0.3(tasmota32)-2_0_13(2023-09-19T11:33:24)
00:00:00.756 SCD40 serial nr 0x0 0x7F 0x0
00:00:00.757 I2C: SCD40 found at 0x62
00:00:01.409 WIF: Connecting to AP1 Simon Channel 6 BSSId 98:DA:C4:A4:46:F2 in mode 11n as tasmota-65DF10-7952...
00:00:01.940 I2C: EZOEC found at 0x61
00:00:01.946 I2C: EZOORP found at 0x63
00:00:01.951 I2C: EZOORP found at 0x64
00:00:01.956 I2C: EZOORP found at 0x65
00:00:01.962 I2C: EZOpH found at 0x66
00:00:04.262 WIF: Connected
00:00:04.520 HTP: Web server active on tasmota-65DF10-7952 with IP address 192.168.10.38
```

Any clues ?  Anything more I can provide to debug?

**Comment by sfromis** [comment, marked as answer]
It is good practice to disable drivers for [I2C sensors](https://tasmota.github.io/docs/I2CDEVICES/) you do not have connected, to avoid risks of address-dependent clashes. You can disable the SCD40 driver using the command `I2CDriver62 0`.

**Reply by SimonFili** [reply]
Thanks, funny, I remember this now, since I debugged EZO on another controller before.
Tried it and worked fine after a soft reset.
Still interesting that the issue was not there on a power on.

**Reply by barbudor** [reply]
The behavior of the EZO might be slightly different between cold and warm boot (because on a warm boot it was already setup) which made the SCD believe that it was an actual SCD chip.
Auto identification of an i2c chip is far from a perfect science and is likely to fail in most of the case. So better only have the driver you need enabled so Tasmota will skip the others even if they are in the binary build.
